examples/async.rs

```diff
@@ -1,30 +1,29 @@
-use async_std::task;
-use failure::Error;
-use hypercore::Feed;
-use random_access_storage::RandomAccess;
-use std::fmt::Debug;
-
-async fn append<T>(feed: &mut Feed<T>, content: &[u8])
-where
-    T: RandomAccess<Error = Error> + Debug,
-{
-    feed.append(content).unwrap();
-}
-
-async fn print<T>(feed: &mut Feed<T>)
-where
-    T: RandomAccess<Error = Error> + Debug,
-{
-    println!("{:?}", feed.get(0));
-    println!("{:?}", feed.get(1));
-}
-
-fn main() {
-    task::block_on(task::spawn(async {
-        let mut feed = Feed::default();
-
-        append(&mut feed, b"hello").await;
-        append(&mut feed, b"world").await;
-        print(&mut feed).await;
-    }));
-}
+use async_std::task;
+use hypercore::Feed;
+use random_access_storage::RandomAccess;
+use std::fmt::Debug;
+
+async fn append<T>(feed: &mut Feed<T>, content: &[u8])
+where
+    T: RandomAccess<Error = Box<dyn std::error::Error + Send + Sync>> + Debug,
```

Comment:
Maybe define an alias in the crate root?

Reply:
I've wanted to do that, but `trait alias` are only available on nightly:

```
error[E0658]: trait aliases are experimental
 --> examples\async.rs:6:1
  |
6 | trait Storage = RandomAccess<Error = Box<dyn std::error::Error + Send + Sync>> + Debug + Send;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: for more information, see https://github.com/rust-lang/rust/issues/41517
```

Reply:
I'll open an issue to track https://github.com/rust-lang/rust/issues/41517 and move it on after this lands.